src/config.py

```diff
@@ -124,95 +87,83 @@ def decode_and_parse(value) -> Optional[Dict]:
     return json.loads(decoded)
 
 
+def verify_path_is_directory(path):
+    if not path.is_dir():
+        raise ValueError(f"Invalid folder path: '{path}', not a directory!")
+    return path
+
+
 class FunctionConfig(BaseModel):
-    external_id: str
-    folder_path: Path
-    common_folder_path: Optional[Path]
-    file: str
-    schedule_file: Optional[str]
-    data_set_external_id: Optional[str]
-    secret: Optional[str]
+    function_name: non_empty_str
+    function_folder: Path
+    function_secrets: non_empty_str = None
+    function_file: constr(min_length=1, strip_whitespace=True, regex=r"^[\w\- ]+\.py$")  # noqa: F722
+    schedule_file: constr(min_length=1, strip_whitespace=True, regex=r"^[\w\- /]+\.ya?ml$") = None  # noqa: F722
+    data_set_external_id: non_empty_str = None
+    common_folder: Path = None
     tenant: TenantConfig
-    overwrite: bool
     remove_only: bool = False
-    deploy_wait_time_sec: int = 1200  # 20 minutes
-    cpu: Optional[float]
-    memory: Optional[float]
-    owner: Optional[str]
-
-    @validator("file")
-    def valid_file(cls, value):
-        if not value.endswith(".py"):
-            raise ValueError(f"Invalid file name, must end with '.py', but got '{value}'")
-        return value
+    cpu: float = None
```

Comment:
Isn't this the same as `Optional[float]`? Why one over the other?

Reply:
Yeah, I have no favorite, but in this case, for readability, I wanted to avoid the following:
```py
schedule_file: Optional[constr(min_length=1, strip_whitespace=True, regex=r"^[\w\- /]+\.ya?ml$")] # noqa: F722

Reply:
(...and so, to not mix, I voted to use `= None` for all)